dts/bindings/sensor/invensense,icm42605.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2020 TDK Invensense
+# SPDX-License-Identifier: Apache-2.0
+
+description: ICM-42605 motion tracking device
+
+# ICM-42605 is SPI.
+compatible: "invensense,icm42605"
+
+include: spi-device.yaml
+
+properties:
+    spi-max-frequency:
+      type: int
+      required: true
+      description: Maximum clock frequency of device's SPI interface in Hz
+
+    int-gpios:
+      type: phandle-array
+      required: true
+      description: |
+        The INT signal default configuration is active-high.  The
+        property value should ensure the flags properly describe the
+        signal that is presented to the driver.
+
+    cpha:
+      type: boolean
+      required: true
+      description: |
+        Set to indicate phase starts with asserted half-phase (CPHA=1).
+        For this driver using this property requires also using cpol.
+
+    cpol:
+      type: boolean
+      required: true
+      description: |
+        Set to indicate clock leading edge is falling (CPOL=1).
+        For this driver using this property requires also using cpha.
+
+    accel-default-hz:
```

Comment:
You can name this field `accel-hz` and use the `default` property to assign a default, so that DT entries not specifying this option will get the device default. According to the datasheet Hz/FS values can take limited values, so the driver should reflect that. In case of DT, adding an `enum` with accepted values is what is usually done.

![image](https://user-images.githubusercontent.com/25011557/95179866-82d4d180-07c1-11eb-893e-78e12b3a5ef3.png)

I think that giving freedom on these values may lead to confusions on the capabilities of the device.

Reply:
I cannot tested 2K to 8K Hz on Nordic board due to hardware limitation.
Anyway, your suggestion is correct and I will modify after test with Logic analyzer.

Reply:
@gmarull Tested From 1K to 8K, without result printing.
With result printing, 1K is maximum frequency for accel and gyro